lib/mzgl/gl/api/OpenGL/OpenGLVbo.h

```diff
@@ -0,0 +1,267 @@
+#pragma once
+#include "Vbo.h"
+#include "mzOpenGL.h"
+
+static int primitiveTypeToGLMode(Vbo::PrimitiveType mode) {
+	switch (mode) {
+		case Vbo::PrimitiveType::Triangles: return GL_TRIANGLES;
+		case Vbo::PrimitiveType::TriangleStrip: return GL_TRIANGLE_STRIP;
+		case Vbo::PrimitiveType::LineStrip: return GL_LINE_STRIP;
+		case Vbo::PrimitiveType::Lines: return GL_LINES;
+
+		default: {
+			Log::e() << "ERROR!! invalid primitive type " << (int) mode;
+			return GL_TRIANGLES;
+		}
+	}
+}
+#include "OpenGLShader.h"
+
+class OpenGLVbo : public Vbo {
+public:
+	//~OpenGLVboImpl() override { deallocate(); }
```

Comment:
If this is inheriting it will need a virtual destruction even if its just =default;